app/experimenter/nimbus-ui/src/components/FormAudience/index.tsx

```diff
@@ -2,76 +2,178 @@
   isMissingField: (fieldName: string) => boolean;
+  isLoading: boolean;
+  onSubmit: (data: Record<string, any>, reset: Function) => void;
+  onNext?: (ev: React.FormEvent) => void;
 }) => {
-  const {
-    channels,
-    firefoxMinVersion,
-    targetingConfigSlug,
-    populationPercent,
-    totalEnrolledClients,
-    proposedEnrollment,
-    proposedDuration,
-  } = experiment;
+  const config = useConfig();
 
   const channelsOptions = config.channels?.filter(
     (item): item is getConfig_nimbusConfig_channels => item !== null,
   );
-  const channelsDefaultValue = channels?.map((channel) =>
+  const channelsDefaultValue = experiment.channels?.map((channel) =>
     channelsOptions?.find((option) => option?.value === channel),
   );
 
+  const defaultValues = {
+    channels: channelsDefaultValue,
+    firefoxMinVersion: experiment.firefoxMinVersion || "",
+    targetingConfigSlug: experiment.targetingConfigSlug || "",
+    populationPercent: experiment.populationPercent || 0,
+    totalEnrolledClients: experiment.totalEnrolledClients,
+    proposedEnrollment: experiment.proposedEnrollment || 7,
+    proposedDuration: experiment.proposedDuration || 28,
+  };
+
+  const {
+    handleSubmit,
+    register,
+    control,
+    reset,
+    errors,
+    formState: { isValid, isSubmitted, touched },
+  } = useForm({
+    mode: "onTouched",
+    defaultValues,
+  });
+
+  type DefaultValues = typeof defaultValues;
+  type DefaultValuesKey = keyof DefaultValues;
+
+  const handleSubmitAfterValidation = useCallback(
+    (dataIn: DefaultValues) => {
+      if (isLoading) return;
+      onSubmit(
+        {
+          ...dataIn,
+          channels: dataIn.channels?.map((item) => item?.value),
+        },
+        reset,
+      );
+    },
+    [isLoading, onSubmit, reset],
+  );
+
+  const handleNext = useCallback(
+    (ev: React.FormEvent) => {
+      ev.preventDefault();
+      onNext!(ev);
+    },
+    [onNext],
+  );
+
+  const fieldValidity = <K extends DefaultValuesKey>(name: K) => ({
+    isInvalid: Boolean(submitErrors[name] || (touched[name] && errors[name])),
+    isValid: Boolean(!submitErrors[name] && touched[name] && !errors[name]),
+  });
+
+  const formControlCommon = <K extends DefaultValuesKey>(
+    name: K,
+    validateRules: ValidationRules = {
+      required: "This field may not be blank.",
+    },
+  ) => ({
+    name,
+    "data-testid": name,
+    ref: register(validateRules),
+    defaultValue: defaultValues[name],
+    ...fieldValidity(name),
+  });
+
+  const FormErrors = <K extends DefaultValuesKey>({ name }: { name: K }) => (
+    <>
+      {errors[name] && (
+        <Form.Control.Feedback type="invalid" data-for={name}>
+          {(errors[name] as FieldError).message}
+        </Form.Control.Feedback>
+      )}
+      {submitErrors[name] && (
+        <Form.Control.Feedback type="invalid" data-for={name}>
+          {submitErrors[name]}
+        </Form.Control.Feedback>
+      )}
+    </>
+  );
```

Comment:
I suspect that nearly all of this form validation & submit utility code could be extracted into a shared module and used on multiple forms. Maybe a function that accepts `defaultValues` and returns at least `FormErrors` and `formControlCommon` defined within a good closure.